deep_daze/deep_daze.py

```diff
@@ -229,24 +230,49 @@ def __init__(
         ).cuda()
 
         self.model = model
-
         self.scaler = GradScaler()
         self.optimizer = Adam(model.parameters(), lr)
         self.gradient_accumulate_every = gradient_accumulate_every
         self.save_every = save_every
-
-        self.text = text
-        textpath = self.text.replace(' ', '_')
-
-        self.textpath = textpath
-        self.filename = Path(f'./{textpath}.png')
+        self.save_date_time = save_date_time
+        self.open_folder = open_folder
         self.save_progress = save_progress
-
+        self.text = text
+        self.textpath = text.replace(" ", "_")
+        self.filename = self.image_output_path()
         self.encoded_text = tokenize(text).cuda()
 
-        self.open_folder = open_folder
-
-    def train_step(self, epoch, i):
+    def image_output_path(self, current_iteration: int = None) -> Path:
```

Comment:
I've tried to be pretty thorough with this code. I've run it multiple times and tried to find every bug. Please give it a thorough look though when you have time.